modules/cli/src/main/scala/scala/cli/commands/ScalaCommand.scala

```diff
@@ -344,6 +353,11 @@ abstract class ScalaCommand[T <: HasGlobalOptions](implicit myParser: Parser[T],
     val logger = options.logging.logger
     val shouldSuppressExperimentalWarning =
       options.globalSuppressWarning.suppressExperimentalFeatureWarning
+        .orElse {
+          getConfigDbOpt(logger)
+            .flatMap(_.getOpt(Keys.suppressExperimentalFeatureWarning, logger))
+        }
+        .getOrElse(false)
```

Comment:
why don't use `shouldSuppressExperimentalFeatureWarnings` in this code? because I see that there is a double logic to extracting suppress from options and config.

Reply:
good catch, fixed